pkg/cluster/api/cdcapi.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pingcap/tiflow/cdc/model"
+	logprinter "github.com/pingcap/tiup/pkg/logger/printer"
+	"github.com/pingcap/tiup/pkg/utils"
+)
+
+// CDCOpenAPIClient is client for access TiCDC Open API
+type CDCOpenAPIClient struct {
+	addrs      []string
+	tlsEnabled bool
+	client     *utils.HTTPClient
+	ctx        context.Context
+}
+
+// NewCDCOpenAPIClient return a `CDCOpenAPIClient`
+func NewCDCOpenAPIClient(ctx context.Context, addrs []string, timeout time.Duration, tlsConfig *tls.Config) *CDCOpenAPIClient {
+	enableTLS := false
+	if tlsConfig != nil {
+		enableTLS = true
+	}
+
+	return &CDCOpenAPIClient{
+		addrs:      addrs,
+		tlsEnabled: enableTLS,
+		client:     utils.NewHTTPClient(timeout, tlsConfig),
+		ctx:        ctx,
+	}
+}
+
+func drainCapture(client *CDCOpenAPIClient, target string) (int, error) {
+	api := "/api/v1/captures/drain"
+	endpoints := client.getEndpoints(api)
+
+	request := model.DrainCaptureRequest{
+		CaptureID: target,
+	}
+	body, err := json.Marshal(request)
+	if err != nil {
+		return 0, err
+	}
+
+	var data []byte
+	_, err = tryURLs(endpoints, func(endpoint string) ([]byte, error) {
+		body, statusCode, err := client.client.PUT(client.ctx, endpoint, bytes.NewReader(body))
+		if err != nil {
+			if statusCode == http.StatusNotFound {
+				// old version cdc does not support `DrainCapture`, return nil to trigger hard restart.
+				return data, nil
+			}
+
+			if bytes.Contains(body, []byte("scheduler request failed")) {
```

Comment:
If tiup relies the magic string, we need to add a unit test to ticdc to make sure the string never change after since.

Reply:
the magic string is a part of CDC RFC ERROR, https://github.com/pingcap/tiflow/blob/master/pkg/errors/cdc_errors.go#L51-L58

I think it's covered in https://github.com/pingcap/tiflow/blob/master/cdc/api/v1/api_test.go#L473, we can enhance this later.